SConstruct

```diff
@@ -189,6 +190,12 @@ if env['distcc']:
 
 if env['ccache']: env.Tool('ccache')
 
+if env['compile_db']:
+    env.Tool('compilation_db')
+    cdb = env.CompilationDatabase()
+    env['COMPILATIONDB_PATH_FILTER'] = "build/" + env["build"] + "/[!m][!o][!d]*"
```

Comment:
https://docs.python.org/3/library/fnmatch.html

The issue is that the modules directory will fail many clang-tidy checks in ways that aren't fixable (since it's not our code). I could remove the filter, but then `scons` won't generate a compilation database usable by `clang-tidy`.

Reply:
I replaced this with a regex on the `run-clang-tidy` command itself.